src/peer_connection_handle.cpp

```diff
@@ -0,0 +1,314 @@
+	pc->disconnect(ec, op, error);
+}
+
+bool peer_connection_handle::is_disconnecting() const
+{
+	boost::shared_ptr<peer_connection> pc = native_handle();
+	TORRENT_ASSERT(pc);
+	return pc->is_disconnecting();
+}
+
+bool peer_connection_handle::is_connecting() const
+{
+	boost::shared_ptr<peer_connection> pc = native_handle();
+	TORRENT_ASSERT(pc);
+	return pc->is_connecting();
+}
+
+bool peer_connection_handle::is_outgoing() const
+{
+	boost::shared_ptr<peer_connection> pc = native_handle();
+	TORRENT_ASSERT(pc);
+	return pc->is_outgoing();
+}
+
+bool peer_connection_handle::on_local_network() const
+{
+	boost::shared_ptr<peer_connection> pc = native_handle();
+	TORRENT_ASSERT(pc);
+	return pc->on_local_network();
+}
+
+bool peer_connection_handle::ignore_unchoke_slots() const
+{
+	boost::shared_ptr<peer_connection> pc = native_handle();
+	TORRENT_ASSERT(pc);
+	return pc->ignore_unchoke_slots();
+}
+
+bool peer_connection_handle::failed() const
+{
+	boost::shared_ptr<peer_connection> pc = native_handle();
+	TORRENT_ASSERT(pc);
+	return pc->failed();
+}
+
+#ifndef TORRENT_DISABLE_LOGGING
+void peer_connection_handle::peer_log(peer_log_alert::direction_t direction
+	, char const* event, char const* fmt, ...) const
+{
+	boost::shared_ptr<peer_connection> pc = native_handle();
+	TORRENT_ASSERT(pc);
+	va_list v;
+	va_start(v, fmt);
+	pc->peer_log(direction, event, fmt, v);
+	va_end(v);
+}
+#endif
+
+bool peer_connection_handle::can_disconnect(error_code const& ec) const
+{
+	boost::shared_ptr<peer_connection> pc = native_handle();
+	TORRENT_ASSERT(pc);
+	return pc->can_disconnect(ec);
+}
+
+bool peer_connection_handle::has_metadata() const
+{
+	boost::shared_ptr<peer_connection> pc = native_handle();
+	TORRENT_ASSERT(pc);
+	return pc->has_metadata();
+}
+
+bool peer_connection_handle::in_handshake() const
+{
+	boost::shared_ptr<peer_connection> pc = native_handle();
+	TORRENT_ASSERT(pc);
+	return pc->in_handshake();
+}
+
+void peer_connection_handle::send_buffer(char const* begin, int size, int flags)
+{
+	boost::shared_ptr<peer_connection> pc = native_handle();
+	TORRENT_ASSERT(pc);
+	pc->send_buffer(begin, size, flags);
+}
+
+time_t peer_connection_handle::last_seen_complete() const
+{
+	boost::shared_ptr<peer_connection> pc = native_handle();
+	TORRENT_ASSERT(pc);
+	return pc->last_seen_complete();
+}
+
+time_point peer_connection_handle::time_of_last_unchoke() const
+{
+	boost::shared_ptr<peer_connection> pc = native_handle();
+	TORRENT_ASSERT(pc);
+	return pc->time_of_last_unchoke();
+}
+
+bool bt_peer_connection_handle::packet_finished() const
+{
+	boost::shared_ptr<bt_peer_connection> pc = native_handle();
+	TORRENT_ASSERT(pc);
+	return pc->packet_finished();
+}
+
+bool bt_peer_connection_handle::support_extensions() const
+{
+	boost::shared_ptr<bt_peer_connection> pc = native_handle();
+	TORRENT_ASSERT(pc);
+	return pc->support_extensions();
+}
+
+#if !defined(TORRENT_DISABLE_ENCRYPTION) && !defined(TORRENT_DISABLE_EXTENSIONS)
```

Comment:
what I've done with other public portions of the API is to (attempt to) make it not depend on build configuration. basically, not use the TORRENT_DISABLE_* macros in public header files, but instead just provide empty functions for features that are disabled. I get the impression people prefer that approach (since they don't have to define the same configuration macros when linking against libtorrent)